backend/rpc-impl/src/main/java/org/sonarsource/sonarlint/core/rpc/impl/SonarLintRpcServerImpl.java

```diff
@@ -228,13 +230,19 @@ public CompletableFuture<Void> shutdown() {
         }
       }
       ThreadJobProcessorPool.getProcessors().forEach(JobProcessor::finish);
-      launcherFuture.cancel(true);
+      stopListeningClient();
       return null;
     });
     executor.shutdown();
     return future;
   }
 
+  private void stopListeningClient() {
+    ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
+    executorService.schedule(() -> clientListener.cancel(true), 500, TimeUnit.MILLISECONDS);
```

Comment:
In fact, I wonder if `clientListener.cancel(true)` is the right thing to do. By quickly checking projects using lsp4j, no-one does that. This is maybe what should be fixed

Reply:
Regarding the lsp4j [docs](https://github.com/eclipse-lsp4j/lsp4j/blob/main/documentation/README.md#launch-and-connect-with-a-languageclient), `future.cancel(true)` is the recommended way to stop listening incoming messages.